game-core/src/main/java/tools/image/DecorationPlacer.java

```diff
@@ -181,579 +172,605 @@ public static void main(final String[] args) {
-        centers = PointFileReaderWriter.readOneToOne(is);
-      } catch (final IOException e) {
-        ToolLogger.error("Something wrong with Centers file", e);
-        System.exit(0);
+  private final class DecorationPlacerFrame extends JFrame {
+    private static final long serialVersionUID = 6385408390173085656L;
+
+    // The map image will be stored here
+    private Image image;
+    // hash map for image points
+    private Map<String, List<Point>> currentPoints = new HashMap<>();
+    // hash map for center points
+    private Map<String, Point> centers = new HashMap<>();
+    // hash map for polygon points
+    private Map<String, List<Polygon>> polygons = new HashMap<>();
+    private final JLabel locationLabel = new JLabel();
+    private File currentImageFolderLocation = null;
+    private File currentImagePointsTextFile = null;
+    private Point currentMousePoint = new Point(0, 0);
+    private Triple<String, Image, Point> currentSelectedImage = null;
+    private Map<String, Tuple<Image, List<Point>>> currentImagePoints = new HashMap<>();
+    private boolean highlightAll = false;
+    private boolean createNewImageOnRightClick = false;
+    private Image staticImageForPlacing = null;
+    private boolean showFromTopLeft = true;
+    private ImagePointType imagePointType = ImagePointType.decorations;
+    private boolean cheapMutex = false;
+    private boolean showPointNames = false;
+
+    DecorationPlacerFrame(final String mapName) throws IOException {
+      super("Decoration Placer");
+      setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+      setLocationRelativeTo(null);
+      highlightAll = false;
+      File fileCenters = null;
+      if (mapFolderLocation != null && mapFolderLocation.exists()) {
+        fileCenters = new File(mapFolderLocation, "centers.txt");
       }
-    } else {
-      try {
-        ToolLogger.info("Select the Centers file");
-        final String centerPath = new FileOpen("Select A Center File", mapFolderLocation, ".txt").getPathString();
-        if (centerPath != null) {
-          ToolLogger.info("Centers : " + centerPath);
-          try (InputStream is = new FileInputStream(centerPath)) {
-            centers = PointFileReaderWriter.readOneToOne(is);
+      if (fileCenters == null || !fileCenters.exists()) {
+        fileCenters = new File(new File(mapName).getParent() + File.separator + "centers.txt");
+      }
+      if (fileCenters.exists() && JOptionPane.showConfirmDialog(new JPanel(),
+          "A centers.txt file was found in the map's folder, do you want to use the file to supply the territories "
+              + "centers?",
+          "File Suggestion", 1) == 0) {
+        try (InputStream is = new FileInputStream(fileCenters.getPath())) {
+          ToolLogger.info("Centers : " + fileCenters.getPath());
+          centers = PointFileReaderWriter.readOneToOne(is);
+        } catch (final IOException e) {
+          ToolLogger.error("Something wrong with Centers file");
```

Comment:
The previous version logged the exception, is the exception going to be logged anyways? If so, is this message "something wrong" useful, would it be a duplicate?

Reply:
Yes, the exception is re-thrown, caught in `run()`, and logged there.  I removed the exception here to avoid it being logged twice.

I'm not sure if the log message here is useful or not.  I didn't go through all the code paths where I replaced `System#exit()` with a re-thrown exception to determine if the exception message provided enough context, and so erred on the side of the original author.  There are definitely some messages in these cases that provide useful information (e.g. the name of the file that was being processed), which might not be present in the exception message due to the use of streams.

Reply:
If i had to bet, I'd go with removal; I'm okay with leaving it as it is. Thank you for the second look :+1: